tests/test_json_schema.py

```diff
@@ -5384,3 +5384,16 @@ class MyDataclass:
     json_schema = adapter.json_schema()
     for key in 'abcdef':
         assert json_schema['properties'][key] == {'title': key.upper(), 'type': 'integer'}  # default is not present
+
+
+def test_enum_complex_value() -> None:
+    """https://github.com/pydantic/pydantic/issues/7045"""
+
+    class MyEnum(Enum):
+        foo = (1, 2)
+        bar = (2, 3)
+
+    ta = TypeAdapter(MyEnum)
+
+    # insert_assert(ta.json_schema())
+    assert ta.json_schema() == {'enum': [[1, 2], [2, 3]], 'title': 'MyEnum', 'type': 'array'}
```

Comment:
This would still fail to parse with JSON data matching this schema:

```
TypeAdapter(Direction).validate_json('[0,1]')
#> Input should be (0, 1),(1, 1),(1, 0),(1, -1),(0, -1),(-1, -1),(-1, 0) or (-1, 1) [type=enum, input_value=[0, 1], input_type=list]
```
But serializing complex enums by their value isn't a good idea anyway, as I mentioned in https://github.com/pydantic/pydantic/issues/7045

Reply:
Yup not much we can do about that

Reply:
I think it would work with an appropriate validator, perhaps annoying that you have to write it, but I think given how rare this is it's not unreasonable, and I think this change to the JSON schema generation is an improvement either way.